docs/upgrade-to-v5.md

```diff
@@ -79,8 +79,27 @@ We have set our typescript compilation target to `es2020` which aligns with our
 > **TL;DR**: TODO
 
 TODO(NODE-4771): serializeFunctions bug fix makes function names outside the ascii range get serialized correctly
-> This will preserve newer ECMAScript 2020 features like optional chaining, nullish coalescing, export * as ns, and dynamic import(...) syntax. It also means bigint literals now have a stable target below esnext.
 
 ### Remove `Map` export
 
 This library no longer polyfills [ES Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) and the export "Map" was removed. Users should migrate to using the global Map constructor available in all supported JS environments.
+
+### Remove deprecated ObjectId methods
+
+The following deprecated methods have been removed:
+
+- `ObjectId.prototype.generate`
+  - Instead, generate a new ObjectId with the constructor: `new ObjectId()` or using the `static generate(time?: number)` method.
+- `ObjectId.prototype.generationTime`
+  - Instead, use `static createFromTime()` and `getTimestamp()` to set and inspect these values on an `ObjectId()`
+
+- `ObjectId.prototype.getInc`
+- `ObjectId.prototype.get_inc`
+- `ObjectId.get_inc`
+  - The `static getInc()` is private since invoking it increments the next `ObjectId` index, instead users should inspect the counter on newly created ObjectId if it is needed.
+  - See the following code snippet for how to obtain the counter field of an ObjectId:
+
+```ts
+const oid = new ObjectId();
+const counterField = oid.id[9] << 16 | oid.id[10] << 8 | oid.id[11];
```

Comment:
completely optional - should we consider just adding a simple getter that does this for users?

```typescript
  getCounter() : number { return this.id[9] << 16 | this.id[10] << 8 | this.id[11]; }
```

Reply:
Per the spec we should not have one: https://github.com/mongodb/specifications/blob/master/source/objectid.rst#counter